modepy/tools.py

```diff
@@ -491,6 +491,8 @@ def reshape_array_for_tensor_product_space(
         function along a given dimension will be represented by variation
         of array entries along the corresponding array axis.
     """
+    if space.spatial_dim == 1:
```

Comment:
```suggestion
    if len(space.bases) == 1:
```
This handles e.g. a trivial tensor product of triangle (2D).

Reply:
This would fail for the above reason, i.e. in the 1D case `PN` does not have a `bases` attribute.

Reply:
If we drop support for being able to create 1D tensor-product element groups and discretizations in upstream packages, then we won't need this change.

Reply:
Sorry, I clearly had not read the PR description. In that case, the first thing that needs fixing is the type annotation of this function. And next, the proper spelling of this test is `if not isinstance(space, TensorProductSpace)`.

Analogous below, I think.